tests/lib/modem_trace/modem_trace_rtt/Kconfig

```diff
@@ -0,0 +1,23 @@
+config HAS_SEGGER_RTT
+	bool
+	default y
+	help
+	    This symbol overrides the promptless symbol HAS_SEGGER_RTT because this is needed by
+	    the unit test.
+
+choice NRF_MODEM_LIB_TRACE_MEDIUM
+	prompt "Test: nRF modem trace medium"
+	default NRF_MODEM_LIB_TRACE_MEDIUM_RTT
+	help
+	    This symbols overrides the symbol NRF_MODEM_LIB_TRACE_MEDIUM in order to override
+		the latter's dependencies on NRF_MODEM_LIB and its dependencies.
+endchoice
```

Comment:
Why are you overwriting the Kconfig for those ?

You should instead add:
```
CONFIG_NRF_MODEM_LIB_TRACE_ENABLED=y
CONFIG_NRF_MODEM_LIB_TRACE_MEDIUM_RTT=y
```
to the `prj.conf` file for this test.

```suggestion
```

Reply:
alternatives has been discussed offline, work will be carried out to establish some unit test and mock guidelines in general. 

But that is considered outside the scope of this PR, so accepting this for now.

No changes requested.